catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/test/java/org/codice/ddf/catalog/subscriptionstore/SubscriptionCacheLoaderTest.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.subscriptionstore;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang.Validate.notNull;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.codice.ddf.catalog.subscriptionstore.common.CachedSubscription;
+import org.codice.ddf.catalog.subscriptionstore.common.SubscriptionMetadata;
+import org.codice.ddf.catalog.subscriptionstore.common.SubscriptionPersistor;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SubscriptionCacheLoaderTest {
+
+    @Mock
+    private SubscriptionPersistor mockPersistor;
+
+    @Mock
+    private SubscriptionMetadata mockMetadata;
+
+    private SubscriptionCacheLoader cacheLoader;
+
+    @Before
+    public void setup() {
+        cacheLoader = new SubscriptionCacheLoader(mockPersistor);
+    }
+
+    @Test
+    public void testLoad() {
+        Map<String, SubscriptionMetadata> results = ImmutableMap.of("id", mockMetadata);
+        when(mockPersistor.getSubscriptions()).thenReturn(results);
+        CachedSubscription cachedSubscription = cacheLoader.load("id");
+        assertThat(cachedSubscription.getMetadata(), is(mockMetadata));
+    }
+
+    @Test
+    public void testLoadReturnsNull() {
+        Map<String, SubscriptionMetadata> results = ImmutableMap.of("id", mockMetadata);
+        when(mockPersistor.getSubscriptions()).thenReturn(results);
+        CachedSubscription cachedSubscription = cacheLoader.load("not_id");
+        assertThat(cachedSubscription, is(nullValue()));
+    }
+
```

Comment:
❓  For the 3 loadAll tests, should we put a comment like the one on the loadAll method explaining why the keys don't matter?